prometheus/getmetrics_prometheus.py

```diff
@@ -0,0 +1,360 @@
+        metric_data_all.append(value_map)
+    except requests.exceptions.ConnectionError:
+        logger.error("Unable to connect to: " + url)
+    except requests.exceptions.MissingSchema as e:
+        logger.error("Missing Schema: " + str(e))
+    except requests.exceptions.Timeout:
+        logger.error("Timed out connecting to: " + url)
+    except requests.exceptions.TooManyRedirects:
+        logger.error("Too many redirects to: " + url)
+    except ValueError:
+        logger.error("Unable to parse result from: " + url)
+    except requests.exceptions.RequestException as e:
+        logger.error(str(e))
+    return metric_data_all
+
+
+def send_data(metric_data):
+    send_data_time = time.time()
+    # prepare data for metric streaming agent
+    to_send_data_dict = {"metric_data": json.dumps(metric_data),
+                         "licenseKey": agent_config_vars['licenseKey'],
+                         "projectName": agent_config_vars['projectName'],
+                         "userName": agent_config_vars['userName'],
+                         "instanceName": socket.gethostname().partition(".")[0],
+                         "samplingInterval": str(int(reporting_config_vars['reporting_interval'] * 60)),
+                         "agentType": "prometheus"}
+
+    to_send_data_json = json.dumps(to_send_data_dict)
+    logger.debug("TotalData: " +
+                 str(len(bytearray(to_send_data_json))) + " Bytes")
+
+    # send the data
+    post_url = parameters['serverUrl'] + "/customprojectrawdata"
+    response = requests.post(post_url, data=json.loads(to_send_data_json))
+    if response.status_code == 200:
+        logger.info(str(len(bytearray(to_send_data_json))) +
+                    " bytes of data are reported.")
+    else:
+        logger.info("Failed to send data.")
+    logger.debug("--- Send data time: %s seconds ---" %
+                 (time.time() - send_data_time))
+
+
+def chunks(l, n):
+    """Yield successive n-sized chunks from l."""
+    for i in range(0, len(l), n):
+        yield l[i:i + n]
+
+
+def set_logger_config(level):
+    """Set up logging according to the defined log level"""
+    # Get the root logger
+    logger_obj = logging.getLogger(__name__)
+    # Have to set the root logger level, it defaults to logging.WARNING
+    logger_obj.setLevel(level)
+    # route INFO and DEBUG logging to stdout from stderr
+    logging_handler_out = logging.StreamHandler(sys.stdout)
+    logging_handler_out.setLevel(logging.DEBUG)
+    # create a logging format
+    formatter = logging.Formatter('%(asctime)s - %(name)s - %(process)d - %(threadName)s - %(levelname)s - %(message)s')
+    logging_handler_out.setFormatter(formatter)
+    logger_obj.addHandler(logging_handler_out)
+
+    logging_handler_err = logging.StreamHandler(sys.stderr)
+    logging_handler_err.setLevel(logging.WARNING)
+    logger_obj.addHandler(logging_handler_err)
+    return logger_obj
+
+
+if __name__ == "__main__":
+    GROUPING_START = 15000

```

Comment:
isnt this supposed to come from config?

Reply:
The start range is fixed for an agent type. We take this as base